vault/expiration.go

```diff
@@ -193,11 +193,13 @@ func (m *ExpirationManager) revokeCommon(leaseID string, force bool) error {
 	}
 
 	// Revoke the entry
-	if err := m.revokeEntry(le); err != nil {
-		if !force {
-			return err
-		} else {
-			m.logger.Printf("[WARN]: revocation from the backend failed, but in force mode so ignoring; error was: %s", err)
+	if !skipToken || le.Auth == nil {
```

Comment:
It's fine actually. If `le.Auth` is `nil`, then there will be no deadlock issue coz `revokeEntry` will then not invoke token_store at all.

It's looking fine now.